src/crypto.rs

```diff
@@ -556,6 +569,58 @@ impl TryFrom<&serde_cbor::Value> for COSEKey {
     }
 }
 
+impl TryFrom<&X509PublicKey> for COSEKey {
+    type Error = WebauthnError;
+    fn try_from(cert: &X509PublicKey) -> Result<COSEKey, Self::Error> {
+        let ec_key = cert
+            .pubk
+            .public_key()
+            .and_then(|pk| pk.ec_key())
+            .map_err(WebauthnError::OpenSSLError)?;
+        let ec_group = ec_key.group();
+
+        let key = match cert.t {
+            COSEContentType::ECDSA_SHA256
+            | COSEContentType::ECDSA_SHA384
+            | COSEContentType::ECDSA_SHA512 => {
+                let curve = ec_group
+                    .curve_name()
+                    .ok_or(WebauthnError::OpenSSLErrorNoCurveName)
+                    .and_then(ECDSACurve::try_from)?;
+
+                let mut ctx =
+                    openssl::bn::BigNumContext::new().map_err(WebauthnError::OpenSSLError)?;
+                let mut x = openssl::bn::BigNum::new().map_err(WebauthnError::OpenSSLError)?;
+                let mut y = openssl::bn::BigNum::new().map_err(WebauthnError::OpenSSLError)?;
+
+                ec_group
+                    .generator()
+                    .affine_coordinates_gfp(ec_group, &mut x, &mut y, &mut ctx)
+                    .map_err(WebauthnError::OpenSSLError)?;
+
+                // TODO: error variant here
+                use std::convert::TryInto;
+                let x: [u8; 32] = x.to_vec().try_into().unwrap();
+                let y: [u8; 32] = y.to_vec().try_into().unwrap();
+
+                Ok(COSEKeyType::EC_EC2(COSEEC2Key { curve, x, y }))
+            }
+            COSEContentType::RS256
+            | COSEContentType::RS384
+            | COSEContentType::RS512
+            | COSEContentType::PS256
+            | COSEContentType::PS384
+            | COSEContentType::PS512
+            | COSEContentType::EDDSA
+            | COSEContentType::INSECURE_RS1 => Err(WebauthnError::COSEKeyInvalidType),
```

Comment:
Could we worth putting in a error!() log macro here too.

Reply:
For the all of the variants or just `INSECURE_RS1`?

Reply:
Foll all variants, to show that we tried to convert something we don't support. That way people can report the bug easier :)